### PR TITLE
Checkstyle: fix MethodName supplied expression #319

### PR DESCRIPTION
**Pull request by eugenepeh**
#319
Ready for review

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@eugenepeh submitted v1 for review.

* [v1 1/2] [Checkstyle: fix MethodName supplied expression](https://github.com/se-edu/addressbook-level4/pull/366/commits/8cd6983fbce4d74f9d1a426d5d97cc507094513e)
* [v1 2/2] [Checkstyle: fix MethodName rule violations ](https://github.com/se-edu/addressbook-level4/pull/366/commits/595c153c937217d11cf60152e396f394b60a0f5b)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/27cb4621dca1bf474b2838352a90873f3fbcb4ef...595c153c937217d11cf60152e396f394b60a0f5b))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/366/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by eugenepeh** [comment]
Thanks @yamgent, oops yes, there's nothing wrong with the first one.
will change that line to 

"We should enforce a standardization such that only method names that follows this format
"sortList_emptyList_exceptionThrown" are allowed, and others such as
"sortList_EmptyList_ExceptionThrown" (capital after underscore) should prompt an error.

> They are not enforced because the regex is not written to cover these cases.

Originally, there's a supplied regex expression though, just that it is not correct.

**Comment by yamgent** [comment]
> Originally, there's a supplied regex expression though, just that it is not correct.

Oh okay, then you can just mention that `the regex that was enforcing the standard was written incorrectly`. The original statement can be more concise that way.

**Comment by CanIHasReview** [comment]
# v2

@eugenepeh submitted v2 for review.

* [v2 1/1] [Checkstyle: fix MethodName supplied expression](https://github.com/se-edu/addressbook-level4/pull/366/commits/58cb90261262e9994c116f9cdf895ca2d6d0c5c9)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/27cb4621dca1bf474b2838352a90873f3fbcb4ef...58cb90261262e9994c116f9cdf895ca2d6d0c5c9)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/cdee299827193287bf15889d88bf68088799fd05~1...cdee299827193287bf15889d88bf68088799fd05))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/366/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -131,8 +131,8 @@
     <module name="MethodName">
       <!-- Validates identifiers for method names against the supplied expression. -->
       <metadata name="altname" value="MethodName"/>
-      <property name="format" value="^[a-z][a-zA-Z0-9]*(_[a-zA-Z0-9]+)*|_+[A-Z]+(_[A-Z]+)*_+$"/>
-      <property name="severity" value="warning"/>
+      <property name="format" value="^[a-z][a-zA-Z0-9]*(_[a-z][a-zA-Z0-9]+)*$"/>
+      <property name="severity" value="error"/>
```
Comment (PierceAndy) [inline config/checkstyle/checkstyle.xml]: As mentioned elsewhere, this line can be omitted.

**Comment by CanIHasReview** [comment]
# v3

@eugenepeh submitted v3 for review.

* [v3 1/1] [Checkstyle: fix MethodName supplied expression](https://github.com/se-edu/addressbook-level4/pull/366/commits/f69daab720e5ee4097305d5e71c7dc6dd3b90b97)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/a61db1bf27f62cc549eb79be0a4598f2f1994b9b...f69daab720e5ee4097305d5e71c7dc6dd3b90b97)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level4/compare/fa6d2c5a76d0660c014e9504029100e5cf3ef348~1...fa6d2c5a76d0660c014e9504029100e5cf3ef348))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/366/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

File: config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -131,8 +131,7 @@
     <module name="MethodName">
```
Comment (damithc) [inline config/checkstyle/checkstyle.xml]: Commit message: 
Describe the exact problems you are fixing. e.g. multiple underscores and (what else?).
i.e. there should be an exact match between the problems being described and your fix. 
In your case, you are not fixing the full `don't have a fixed standard` problem.

Consider something like this:
```
Test case method names are expected to follow the format 
featureBeingTested_scenario_expectedResult. 

However, following aspects of the above naming convention
are not enforced by checkstyle:
1. only one consecutive underscore between two parts
2. no more than 2 underscores
3. each part should of the name should start with a lower case letter
```

Alternatively, you can describe the full problem, but in that case you have to clarify which part of the problem you are solving.

**Comment by CanIHasReview** [comment]
# v4

@eugenepeh submitted v4 for review.

* [v4 1/1] [Checkstyle: fix MethodName supplied expression](https://github.com/se-edu/addressbook-level4/pull/366/commits/5f9872030ddaa927b8604e181cb8aea9f9d15ad2)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/3291675213b3a08db9653da0722ea8fd374074dc...5f9872030ddaa927b8604e181cb8aea9f9d15ad2)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level4/compare/d62300c32ebe0c77575eed858d27bc4826583584~1...d62300c32ebe0c77575eed858d27bc4826583584))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/366/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Pull request merged**